### DISCUSSION TITLE
can't derive FromRow

### DISCUSSION THREAD

**Discussion by SET001** [rwf2/Rocket#2325] [Questions] [answered]
For some reason, I can't derive `FromRow`. What am I missing?

```rust
use rocket_db_pools::sqlx::FromRow;
use serde::{Serialize, Deserialize};

#[derive(Serialize, Deserialize, Debug, FromRow)]
struct Checklist{
  title: String,
  user_id: i32,
  created_at: String
}
```
the error is:
```
error: cannot find derive macro `FromRow` in this scope
  --> src/main.rs:62:41
   |
62 | #[derive(Serialize, Deserialize, Debug, FromRow)]
   |                                         ^^^^^^^
   |
note: `FromRow` is imported here, but it is only a trait, without a derive macro

**Comment by the10thWiz** [comment]
You likely need to sqlx `derive` feature, which rocket_db_pools does not enable be default.

**Comment by SET001** [comment, marked as answer]
the solution is to enable `sqlx_macros` feature in `rocket_db_pools`
also, even though it is reexported by `rocket_db_pools`, I have to explicitly add `sqlx` in dependencies to make this work

```toml
sqlx = { version = "0.5", features = [ "macros"] }
```

Worth to mention that while there is already 0.6 version of sqlx, it seems like `rocket_db_pools` does not work well to it, so I downgraded to 0.5
